extensions/suspend/src/Api/UserResourceFields.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Suspend\Api;
+
+use Flarum\Api\Context;
+use Flarum\Api\Schema;
+use Flarum\User\User;
+
+class UserResourceFields
+{
+    public function __invoke(): array
+    {
+        return [
+            Schema\Boolean::make('canSuspend')
+                ->get($canSuspend = fn (User $user, Context $context) => $context->getActor()->can('suspend', $user)),
+            Schema\Str::make('suspendReason')
+                ->writable($canSuspend)
+                ->visible($canSuspend),
+            Schema\Str::make('suspendMessage')
```

Comment:
theoratically, yes `->requiredWith` `->requiredOnUpdateWith`

though that is not the behavior prior to this PR so out of scope (main goal is to avoid regressions) https://github.com/flarum/framework/blob/2.x/extensions/suspend/src/Listener/SaveSuspensionToDatabase.php